src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs

```diff
@@ -1039,7 +1039,7 @@ private static int TryAddToAnyCore(BlockingCollection<T>[] collections, T item,
                     return OPERATION_FAILED;
 
                 //If the timeout period was not exhausted and the appropriate operation succeeded.
-                if (collections[index].TryAdd(item)) //case#1
+                if (collections[index].TryAdd(item, 0, externalCancellationToken)) //case#1
```

Comment:
I think this case should not have been caught by the analyzer, because the number of parameters in the original method differs by more than one compared to the suggested overload that takes the token.

There are two additional instances in this same file that I fixed for the same reason: two `TryAdd` and one `TryTake`.

I am fixing them all anyway to prevent the warning from making the build fail, by using the default middle parameter value `millisecondsTimeout=0`.

Do please let me know if I should not pass `externalCancellationToken` and instead I should pass `default` or `CancellationToken.None`.

Reply:
I opened a bug in roslyn-analyzers to address this when I get a chance: https://github.com/dotnet/roslyn-analyzers/issues/3786

Reply:
This should be:
```C#
if (collections[index].TryAdd(item, 0, CancellationToken.None)
```
or default, but not externalCancellationToken.

Reply:
Thanks for the confirmation, fixing it.
Would you mind explaining why this case should not pass `externalCancellationToken`?

Reply:
Because when we get here, we've already performed the wait and now believe there's space available. This call is then effectively instantaneous, just performing the add without waiting. If it falls, we loop around and try the whole thing again.